src/main/scala/mesosphere/marathon/api/TaskKiller.scala

```diff
@@ -94,8 +93,8 @@ class TaskKiller @Inject() (
     appInstances: Map[AbsolutePathId, Seq[Instance]],
     force: Boolean)(implicit identity: Identity): Future[DeploymentPlan] = {
     def scaleApp(app: AppDefinition): AppDefinition = {
-      checkAuthorization(UpdateRunSpec, app)
       appInstances.get(app.id).fold(app) { instances =>
+        checkAuthorization(UpdateRunSpec, app)
```

Comment:
this right here is the fix

Reply:
Man, that must have been hard to track down. Thanks!